tests/integration/cmd_dev_recoverability_test.go

```diff
@@ -0,0 +1,118 @@
+	})
+
+	checkIfDevEnvIsUp := func(url, assertString string) {
+		Eventually(func() string {
+			resp, err := http.Get(fmt.Sprintf("http://%s", url))
+			Expect(err).ToNot(HaveOccurred())
+			defer resp.Body.Close()
+
+			body, _ := io.ReadAll(resp.Body)
+			return string(body)
+		}, 120*time.Second, 15*time.Second).Should(Equal(assertString))
+	}
+
+	Context("starting with empty Directory", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			helper.Chdir(commonVar.Context)
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(BeEmpty())
+		})
+
+		It("should verify developer workflow from empty Directory", func() {
+
+			command := []string{"odo", "init"}
+			_, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				helper.ExpectString(ctx, "Select language")
+				helper.SendLine(ctx, "javascript")
+
+				helper.ExpectString(ctx, "Select project type")
+				helper.SendLine(ctx, "Node.js\n")
+
+				helper.ExpectString(ctx, "Which starter project do you want to use")
+				helper.SendLine(ctx, "nodejs-starter\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("server.js"))
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js", "from updated Node.js")
+			Expect(err).ToNot(HaveOccurred())
+			_, _, _, err = devSession.WaitSync()
+			Expect(err).ToNot(HaveOccurred())
+			// "should update the changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from updated Node.js Starter Application!")
+
+			// "changes are made to the applications"
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from updated Node.js", "from Node.js app v2")
+			_, _, _, err = devSession.WaitSync()
+			Expect(err).ToNot(HaveOccurred())
+			// "should deploy new changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from Node.js app v2 Starter Application!")
+
+			//Copy files to emulate network latency
+			helper.Chdir(commonVar.Context)
+			helper.CopyExample(filepath.Join("source", "latency"), commonVar.Context)
+
+			//Trigger network latency emulation
+			// helper.Cmd("systemctl", "start", "docker").ShouldRun()
+			// helper.Cmd("docker-compose", "up", "-d").ShouldRun()
+			stdout := helper.Cmd("docker", "exec", "server", "client", "tc", "qdisc", "add", "dev", "$(ip", "route", "|", "grep", "default", "|", "sed", "-e", "\"s/^.*dev.//\"", "-e", "\"s/.proto.*//\")", "root", "netem", "delay", "20000ms").ShouldRun().Out()
```

Comment:
As discussed together, this adds latency emulation only to the container inside the docker-compose project, not where the test is running.
The docker-compose project was mostly provided as an example of how we could introduce latency to a given client container talking to a separate server; so not to be used as is. Using qdisc also requires the container to be started with network admin capabilities.
We discussed other alternatives to consider, if not possible to use `tc` in our test environment.